src/business_logic/transaction/internal_objects.rs

```diff
@@ -166,3 +170,246 @@ impl InternalDeploy {
+    pub fn new(
+        class_hash: [u8; 32],
+        max_fee: u64,
+        version: u64,
+        nonce: u64,
+        constructor_calldata: Vec<Felt>,
+        signature: Vec<Felt>,
+        contract_address_salt: Address,
+        chain_id: StarknetChainId,
+    ) -> Result<Self, SyscallHandlerError> {
+        let contract_address = calculate_contract_address(
+            &contract_address_salt,
+            &Felt::from_bytes_be(&class_hash),
+            &constructor_calldata,
+            Address(Felt::zero()),
+        )?;
+
+        Ok(Self {
+            contract_address: Address(contract_address),
+            contract_address_salt,
+            class_hash,
+            constructor_calldata,
+            version,
+            nonce,
+            max_fee,
+            signature,
+            chain_id,
+        })
+    }
+
+    pub fn new_for_testing(
+        contract_class: &ContractClass,
+        max_fee: u64,
+        contract_address_salt: Address,
+        constructor_calldata: Vec<Felt>,
+        chain_id: StarknetChainId,
+        signature: Vec<Felt>,
+    ) -> Result<Self, SyscallHandlerError> {
+        Self::new(
+            felt_to_hash(&compute_class_hash(contract_class).unwrap()),
+            max_fee,
+            TRANSACTION_VERSION,
+            0,
+            constructor_calldata,
+            signature,
+            contract_address_salt,
+            chain_id,
+        )
+    }
+
+    pub fn get_state_selector(&self, _general_config: StarknetGeneralConfig) -> StateSelector {
+        StateSelector {
+            contract_addresses: vec![self.contract_address.clone()],
+            class_hashes: vec![self.class_hash],
+        }
+    }
+
+    fn _apply_specific_concurrent_changes<S>(
+        &self,
+        state: &mut S,
+        general_config: &StarknetGeneralConfig,
+    ) -> Result<TransactionExecutionInfo, StateError>
+    where
+        S: Default + State + StateReader,
+    {
+        let contract_class = state.get_contract_class(&self.class_hash)?;
+
+        state.deploy_contract(self.contract_address.clone(), self.class_hash)?;
+
+        let mut resources_manager = ExecutionResourcesManager::default();
+        let constructor_call_info = self
+            .handle_constructor(
+                contract_class,
+                state,
+                general_config,
+                &mut resources_manager,
+            )
+            .map_err::<StateError, _>(|_| todo!())?;
+
+        let validate_info = self
+            .run_validate_entrypoint(state, &mut resources_manager, general_config)
+            .map_err::<StateError, _>(|_| todo!())?;
+
+        let actual_resources = calculate_tx_resources(
+            resources_manager,
+            &[Some(constructor_call_info.clone()), validate_info.clone()],
+            TransactionType::DeployAccount,
+            state.count_actual_storage_changes(),
+            None,
+        )
+        .map_err::<StateError, _>(|_| todo!())?;
```

Comment:
why is there a todo! ?

Reply:
I tried a few things to be able to return an error, but by adding this error within the returned one I ended up with an enum with infinite size, and by attempting the inverse (add the returned error to this one) I broke code everywhere else, so I left it as is for now.